src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/JJDisputeController.java

```diff
@@ -264,4 +266,34 @@ public ResponseEntity<TicketImageDataJustinDocument> getTicketImageData(
 
 		return new ResponseEntity<TicketImageDataJustinDocument>(jjDisputeService.getTicketImageByTicketNumber(ticketNumber, documentType), HttpStatus.OK);
 	}
+
+	/**
+	 * DELETE endpoint that deletes a specified {@link JJDispute} as well as its associated data
+	 *
+	 * @param id of the {@link JJDispute} to be deleted
+	 * @param ticketNumber of the {@link JJDispute} to be deleted
+	 */
+	@Operation(summary = "Deletes a particular JJDispute record.")
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "Ok. JJ Dispute record deleted."),
+		@ApiResponse(responseCode = "400", description = "Bad Request."),
+		@ApiResponse(responseCode = "404", description = "JJ Dispute record not found. Delete failed."),
+		@ApiResponse(responseCode = "500", description = "Internal Server Error. Delete failed.")
+	})
+	@DeleteMapping("/dispute")
+	public void deleteDispute(
+			@RequestParam(required = false)
+			@Parameter(description = "If specified, will delete the record of the specified jj dispute by this ID. JJ Dispute ID will take precedence if both ID and ticket number supplied")
+			Long jjDisputeId,
+			@RequestParam(required = false)
+			@Pattern(regexp = "[A-Z]{2}\\d{8}")
+			@Parameter(description = "If specified, will delete the record of the specified jj dispute by this TicketNumber. (Format is XX00000000)", example = "AX12345678")
+			String ticketNumber) {
+		logger.debug("DELETE /dispute called with jjDisputeId: {}, and ticketNumber: {}", StructuredArguments.value("jjDisputeId", jjDisputeId), StructuredArguments.value("ticketNumber", ticketNumber));
+		if (jjDisputeId == null && StringUtils.isBlank(ticketNumber)) {
+			throw new IllegalArgumentException("Either ticketNumber or jjDisputeId must be specified.");
+		}
+
+		jjDisputeService.delete(jjDisputeId, ticketNumber);
```

Comment:
Does this service call throw a NoSuchElementException? I'm not seeint but I may have missed it. Just confirming a 404 is thrown when delete record not found.

Reply:
Yeah, you're right. We're not calling the find endpoint to check if the record exists before deleting it. This is a copy/paste mistake and I will remove 404 from the responses.